datashare-index/src/main/java/org/icij/datashare/text/indexing/elasticsearch/ElasticsearchSearcher.java

```diff
@@ -154,19 +159,41 @@ public Indexer.Searcher set(JsonNode jsonQuery) {
         return this;
     }
 
+    @Override
+    public Indexer.Searcher setFromTemplate(String jsonQueryTemplate, String query) {
+        boolean phraseMatch = phraseMatchTemplate.matcher(jsonQueryTemplate).matches();
+
+        int fuzziness = 0;
+        Matcher fuzzinessMatcher = fuzzinessTemplate.matcher(jsonQueryTemplate);
+        if (fuzzinessMatcher.matches()) {
+            fuzziness = Integer.parseInt(fuzzinessMatcher.group(1));
+        }
+
+        String queryString = buildQueryString(query, fuzziness, phraseMatch, "\\\\\"");
+        jsonQueryTemplate = queryTemplate.matcher(jsonQueryTemplate).replaceAll(queryString);
+        final String queryBody = jsonQueryTemplate;
+
+        this.boolQueryBuilder.must(m -> m.withJson(new StringReader(queryBody)));
+        return this;
+    }
+
     @Override
     public Indexer.Searcher with(String query, int fuzziness, boolean phraseMatches) {
+        String queryString = buildQueryString(query, fuzziness, phraseMatches, "\"");
+        this.boolQueryBuilder.must(m -> m.matchAll(ma -> ma));
+        this.boolQueryBuilder.must(m -> m.queryString(qs -> qs.query(queryString)));
+        return this;
+    }
+    private static String buildQueryString(String query, int fuzziness, boolean phraseMatches, String phraseMatchDoubleQuotes) {
```

Comment:
I added the last parameter `String phraseMatchDoubleQuotes` because it seems `qs.query(queryString)` (l. 184) and `m.withJson(new StringReader(queryBody)` (l.176) are parsing the jsonString in different ways. There must be a better way to do this

Reply:
yes cf my other comment I see duplication between BatchSearch attributes and body template.